### PR TITLE
Query Loop: Include a Query Pagination option on the block variations

### LINKED ISSUE #35134: Query Loop: Include a Query Pagination option on the block variations panel
### What

Add the _Query Pagination_ block when selecting a variation.

### Why

When you add the _Query Loop_ block and select a variation:

QueryLoopBlock

You would expect that those would include pagination, but they don't. It's confusing because you can see pagination settings on the block:

QueryLoopBlock2

But once on the live page, it won't show any pages/pagination.

And it's quite tricky to add the _Query Pagination_ block, you need to have the _Query Loop_ block selected, and use the + icon inside of it to be able to search for the _Query Pagination_ one.

---

Originally reported in https://github.com/Automattic/wp-calypso/issues/56501.

### PR DESCRIPTION
**Pull request by carolinan**
## Description

Adds the query pagination block with its inner blocks to the query loop variations.

Closes https://github.com/WordPress/gutenberg/issues/35134

## How has this been tested?

1. Add a query loop.
2. Select the option "Start blank"
3. Test all 4 variations and confirm that the query pagination is added below the post template.

## Types of changes

Enhancement

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/block-library/src/query/variations.js [hunk 1]
```diff
@@ -69,6 +69,15 @@ const variations = [
 				{},
 				[ [ 'core/post-title' ], [ 'core/post-date' ] ],
 			],
+			[
+				'core/query-pagination',
+				{},
```
Comment (ntsekouras) [inline packages/block-library/src/query/variations.js]: We don't need to pass the innerBlocks as they are defined as the default `template` in `QueryPagination` [here](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/query-pagination/edit.js#L20).

**Review by ntsekouras — COMMENTED** [review]
Thanks for working on this Carolina! I left just a small comment 😄

File: packages/block-library/src/query/variations.js (hunk 1 above)
Comment (carolinan) [reply packages/block-library/src/query/variations.js]: Thanks, I have removed them.

**Review by ntsekouras — APPROVED** [review]

**Comment by paaljoachim** [comment]
I assume that the next/preview or 1,2,3 etc pagination should be a part of the Query Loop? So that it would show up automatically. I am not seeing it, and not sure where or how to find it.

**Comment by ntsekouras** [comment]
>I assume that the next/preview or 1,2,3 etc pagination should be a part of the Query Loop? So that it would show up automatically. I am not seeing it, and not sure where or how to find it.

You can always add this block `Query Pagination` inside `Query Loop`(it has to be the parent because it's coupled with a single Query Loop). It might not be among the first six results of the quick inserter, but it's there if you start typing..

Also if you select a Query pattern to start from, then it comes down to whether the pattern includes this block. I guess we could probably have an extra toolbar button or in toolbar block settings an option to add `Query Pagination` at the end of the innerblocks? 🤔

**Comment by paaljoachim** [comment]
> You can always add this block Query Pagination inside Query Loop(it has to be the parent because it's coupled with a single Query Loop). It might not be among the first six results of the quick inserter, but it's there if you start typing..

Thanks. 

Selecting Query Pattern I just looked at some elements to see where they were in relation to each other. I did not even think about tags/categories/pagination etc. Adding inner blocks is a fairly hidden process. It sounds like we need a new issue to make inner blocks in general a more visible and easier to locate process. I will think on it..:)

**Comment by paaljoachim** [comment]
I made an issue for inner blocks here: https://github.com/WordPress/gutenberg/issues/38578

**Pull request merged**